quaint/src/connector/postgres/native/mod.rs

```diff
@@ -499,19 +615,44 @@ impl Queryable for PostgreSql {
             }
         }
 
-        for col in stmt.columns() {
+        let nullables = self.get_nullable_for_columns(&stmt).await?;
+
+        for (idx, (col, nullable)) in stmt.columns().iter().zip(nullables).enumerate() {
             let (typ, enum_name) = infer_type(self, col.type_()).await?;
 
-            columns.push(ParsedRawColumn::new_named(col.name(), typ).with_enum_name(enum_name));
+            if col.name() == "?column?" {
+                let kind = ErrorKind::QueryInvalidInput(format!("Invalid column name '?column?' for index {idx}. Your SQL query must explicitly alias that column name."));
+
+                return Err(Error::builder(kind).build());
+            }
+
+            columns.push(
+                DescribedColumn::new_named(col.name(), typ)
+                    .with_enum_name(enum_name)
+                    // Make fields nullable by default if we can't infer nullability.
+                    .is_nullable(nullable.unwrap_or(true)),
+            );
         }
 
         for param in stmt.params() {
             let (typ, enum_name) = infer_type(self, param).await?;
 
-            parameters.push(ParsedRawParameter::new_named(param.name(), typ).with_enum_name(enum_name));
+            parameters.push(DescribedParameter::new_named(param.name(), typ).with_enum_name(enum_name));
         }
 
-        Ok(ParsedRawQuery { columns, parameters })
+        let enum_names = self
+            .query_raw("SELECT typname FROM pg_type WHERE typtype = 'e';", &[])
+            .await?
+            .into_iter()
+            .flat_map(|row| row.into_single().ok())
+            .flat_map(|v| v.to_string())
+            .collect::<Vec<_>>();
```

Comment:
Send a single query and select oid to do in-memory matching instead of the infer_type function above (to reduce sql calls)